### ISSUE TITLE
Error:- Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 

### ISSUE THREAD

**Issue by chaudharydeepanshu** [rive-app/rive-flutter#253]
### Description
I have migrated to the latest master branch of flutter for one of my apps which was also uses rive but I found that my app is crashing with an error `Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x7712afa500 in tid 32050 (1.ui), pid 26471 (ample.test_rive)` after few seconds the rive animation starts. But this only happens with some assets. So, I tried the teddy example with some edits and found that it also crashed.

### Steps To Reproduce
1. Download the edited teddy animation from here [riv-rev.zip](https://github.com/rive-app/rive-flutter/files/9438732/riv-rev.zip).
2. **Upgrade to flutter master branch.**
3. Import the rive package and above asset into a new flutter project and replace the main.dart code with this:
```dart
import 'package:flutter/material.dart';
import 'package:rive/rive.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: const SizedBox(
        height: 200,
        child: Center(
          child: RiveAnimation.asset(
            "assets/rive/animated_login_screen.riv",
          ),
        ),
      ),
    );
  }
}
```
4. Let the animation run for a few seconds to crash app.
5. See error ⚠️
```
I/Gralloc4(26471): Adding additional valid usage bits: 0x8202000
F/libc    (26471): Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x7712afa500 in tid 32050 (1.ui), pid 26471 (ample.test_rive)
Lost connection to device.
```

#### Source `.riv`/`.rev` file
[riv-rev.zip](https://github.com/rive-app/rive-flutter/files/9438732/riv-rev.zip)

### Expected behavior
Expected the animation to continue without crashing the app.

### Device & Versions
- Device: Android Physical Device (Redmi Note 10 Pro)
- OS: Android 12
- Flutter Version: 
```
Flutter 3.1.0-0.0.pre.2541 • channel master • https://github.com/flutter/flutter.git
Framework • revision f62ef93e06 (3 hours ago) • 2022-08-27 14:43:25 -0400
Engine • revision 9c54176218
Tools • Dart 2.19.0 (build 2.19.0-146.0.dev) • DevTools 2.16.0
```

#### Additional context
I faced this problem only in the master version of flutter.

Also, looking forward to knowing the reason for the crash.

Thank you.

**Comment by chaudharydeepanshu** [comment]
Any update on this?

**Comment by mikeesouth** [comment]
I'm getting this too, unfortunately. I'm on flutter stable (3.7.0) and latest rive (0.10.1). I get the problem on Windows and Android Emulator (API version 33) with Flutter in debug mode but I haven't tested other variants yet (release mode/MacOS/iOS emulator/physical devices etc).

I can load my asset and play the idle state without problems and I can have the app running for 10minutes+ without a crash. If I queue state machine animations - it crashes within seconds with the same error as OP: `Fatal signal 5 (SIGTRAP)` but with `code 128 (SI_KERNEL)`.

I tried re-downloading my asset from the Rive editor and for the latest runtime (v7) but I still get the same error unfortunately. I've tested on two different machines and I have performed a flutter clean.

I'm trying the reproducible example from OP now to see if it also crashes for me on flutter stable (3.7.0). I will also try to downgrade to rive 0.9.1.

**Comment by HayesGordon** [comment]
Hi all!

I believe this to be a compiler issue in Dart. See related issue here: https://github.com/flutter/flutter/issues/118559#issuecomment-1404770385

We're waiting for a Cherry Pick to Flutter stable.

At the moment it should not crash on a profile/release build (for Flutter 3.7.0), or on the Master Flutter branch (where the fix has already landed), or pre 3.7.0

**Comment by mikeesouth** [comment]
This seems fixed for me now, using Flutter 3.7.2 👍

**Comment by HayesGordon** [comment]
That's great!

Closing this issue then. Please re-open if anyone encounters issues or if this pops up again.
